ykrt/src/compile/jitc_yk/jit_ir/parser.rs

```diff
@@ -889,6 +876,27 @@ enum ASTType {
     Void,
 }
 
+/// Hand out X86 registers to JIT tests that want them, mapping them to the DWARF registers that
+/// yksmp uses.
+#[cfg(target_arch = "x86_64")]
+mod x64_regs {
+    pub(super) fn fp_reg_iter() -> impl Iterator<Item = u16> {
+        // FP registers on x64 DWARF are 17 to 32 inclusive
+        17..=32
+    }
+
+    pub(super) fn gp_reg_iter() -> impl Iterator<Item = u16> {
+        // For reasons that are above my pay grade, DWARF uses a different ordering of registers to
+        // everyone else. To lessen the confusion we hand out registers in "Intel order" i.e. the order
+        // that everyone who doesn't read the DWARF spec expects. This array encodes that mapping.
+        let intel_dwarf_mapping: [u16; 14] = [0, 2, 1, 3, 4, 5, 8, 9, 10, 11, 12, 13, 14, 15];
```

Comment:
Just to be sure I understood this: the JIT IR parser now hands out DWARF registers, and this mapping takes an Intel register and returns the appropriate DWARF register. This makes sure the register allocator hands out things in Intel order while using DWARF numbers.

Reply:
Correct!